thriftbp/client_pool.go

```diff
@@ -429,6 +434,19 @@ func NewCustomClientPoolWithContext(
 	return newClientPool(ctx, cfg, genAddr, protoFactory, middlewares...)
 }
 
+const ThriftHostnameHeader = "thrift-hostname"
```

Comment:
is this intentionally lower-case?

for THeaders we use Camel-Case (see https://github.com/reddit/baseplate.go/blob/master/transport/headers.go), it will be better to keep it consistent.

Reply:
that's the value the ingress expects sadly.

@davinci26 can comment on whether this can be changed or not

Reply:
So according to the HTTP RFC header casing doesnt matter. @fishy does the same apply for `thrift`?

>    Each header field consists of a case-insensitive field name followed
   by a colon (":"), optional leading whitespace, the field value, and
   optional trailing whitespace.
   
   In any case `envoy` will lower case the headers when parsing the thrift payload so it doesn't care but it would be nice if we took a dependency on a thrift spec definition rather than an envoy implementation detail.
   
   
   https://github.com/envoyproxy/envoy/blob/12210e53db6cea79b58ee027f047faa356e1e392/source/extensions/filters/network/thrift_proxy/header_transport_impl.cc#L163

Reply:
well this is not http header, it's thrift header.

envoy lower case all headers was a bug and fixed in latest version of envoy: https://github.com/envoyproxy/envoy/issues/20595

Reply:
> well this is not http header, it's thrift header.

I know that, this is what I am asking if you know. Does the thrift protocol state that thrift headers casing is important or not?

Reply:
even if this breaks the code pattern here. This is a header defined externally to baseplate.go so it should be fine to have different casing